nativelib/src/main/scala/scala/scalanative/unsafe/Ptr.scala

```diff
@@ -62,4 +60,12 @@ object Ptr {
 
   @alwaysinline implicit def ptrToCStruct[T <: CStruct](ptr: Ptr[T])(
       implicit tag: Tag[T]): T = !ptr
+
+  @alwaysinline implicit def ptrToCFuncPtr[F <: CFuncPtr](ptr: Ptr[Byte])(
+      implicit tag: Tag.CFuncPtrTag[F]): F =
+    Boxes.boxToCFuncRawPtr(ptr.rawptr).as[F]
+
+  @alwaysinline implicit def cFuncPtrToPtr[T](ptr: CFuncPtr): Ptr[Byte] = {
+    Boxes.boxToPtr[Byte](Boxes.unboxToCFuncRawPtr(ptr))
+  }
```

Comment:
I don't think those two methods should be implicit. Not all `Ptr[Byte]`s are function pointers, especially not of an arbitrary `F`, and so that conversion should be explicit in the users' code.

This is fundamentally different from `ptrToCArray` and `ptrToCStruct` above: for them, the conversion is always safe when it typechecks, and hence it can be implicit.

Reply:
Additionally, I'm not sure this is the best place for these functions. `Ptr` seems too general a place for them. Consider declaring them in `CFuncPtr`, so that they would be called as `CFuncPtr.fromPtr(ptr)` and `CFuncPtr.toPtr(cFuncPtr)`, for example.